evap/static/ts/src/notebook.ts

```diff
@@ -0,0 +1,88 @@
+declare const bootstrap: typeof import("bootstrap");
+import { unwrap, assert } from "./utils.js";
+
+const NOTEBOOK_LOCALSTORAGE_KEY = "evap_notebook_open";
+const COLLAPSE_TOGGLE_BUTTON_ID = "notebookButton";
+const WEBSITE_CONTENT_ID = "evapContent";
+const NOTEBOOK_FORM_ID = "notebook-form";
+
+class NotebookFormLogic {
+    private readonly notebook: HTMLFormElement;
+
+    constructor(notebookFormId: string) {
+        this.notebook = unwrap(document.getElementById(notebookFormId)) as HTMLFormElement;
+    }
+
+    private onSubmit = (event: SubmitEvent): void => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        const submitter = unwrap(event.submitter) as HTMLButtonElement;
+        submitter.disabled = true;
+        this.notebook.setAttribute("data-state", "sending");
+
+        fetch(this.notebook.action, {
+            body: new FormData(this.notebook),
+            method: "POST",
+        })
+            .then(response => {
+                assert(response.ok);
+                this.notebook.setAttribute("data-state", "successful");
+                setTimeout(() => {
+                    this.notebook.setAttribute("data-state", "ready");
+                    submitter.disabled = false;
+                }, 2000);
+            })
+            .catch(() => {
+                this.notebook.setAttribute("data-state", "error");
+                submitter.disabled = false;
+                alert(submitter.dataset.error);
+            });
+    };
+
+    public attach = (): void => {
+        unwrap(this.notebook).addEventListener("submit", this.onSubmit);
+    };
+}
+
+export class NotebookLogic {
+    private readonly notebook: HTMLElement;
+    private readonly content: HTMLElement;
+    private form: NotebookFormLogic;
```

Comment:
I am not sure what these variables are holding exactly, could you give describe them a bit more explicitly?